CHANGELOG.md

```diff
@@ -1,6 +1,13 @@
 Changes by Version
 ==================
 
+0.42.0
```

Comment:
In the release PR, we only add changes that matter to end-users into `CHANGELOG.md`, not all the changes or improvements that don't matter to the end-user. 
Also, you should include `* Bumped OpenTelemetry Collector to v0.42.0  in CHANGELOG. `

Reply:
Which change does not matter? I find image changes useful to know.

Reply:
I mean this isn't end-user related IMO `Parse flags before using them in config ([#662](https://github.com/open-telemetry/opentelemetry-operator/pull/662), [@rubenvp8510](https://github.com/rubenvp8510))`

Reply:
it's a fix that fixes paring of the flags which was broken